app/services/container_dashboard_service.rb

```diff
@@ -119,29 +121,44 @@ def get_url_to_entity(entity)
 
   def heatmaps
     # Get latest hourly rollup for each node.
-    node_ids = @ems.container_nodes if @ems
+    node_ids = @ems.container_nodes if @ems.present?
     metrics = MetricRollup.latest_rollups(ContainerNode.name, node_ids)
+    metrics = metrics.includes(:resource => [:ext_management_system]) unless @ems.present?
+
+    node_cpu_usage = nil
+    node_memory_usage = nil
 
     if metrics.any?
-      {
-        :nodeCpuUsage => metrics.collect { |m|
-          avg_cpu = m.cpu_usage_rate_average.round(2)
-          {
-            :id      => m.resource_id,
-            :tooltip => "#{m.resource.name} - #{avg_cpu}%",
-            :value   => avg_cpu / 100.0 # 1% should be 0.01
-          }
-        },
-        :nodeMemoryUsage => metrics.collect { |m|
-          avg_mem = m.mem_usage_absolute_average.round(2)
-          {
-            :id      => m.resource_id,
-            :tooltip => "#{m.resource.name} - #{avg_mem}%",
-            :value   => avg_mem / 100.0 # 1% should be 0.01
-          }
+      node_cpu_usage = []
+      node_memory_usage = []
+
+      metrics.each do |m|
```

Comment:
The previous code ran ```metrics.collect``` twice but overall I doubt if there will be a situation with more than 200 entries so it could stay as it was before